Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -141,18 +143,24 @@ public function handle(CommandInterface $command): void
      */
     public function projectionState(string $projectionStateClassName): ProjectionStateInterface
     {
+        if (!isset($this->projectionStateCache)) {
+            foreach ($this->projectionsAndCatchUpHooks->projections as $projection) {
+                if ($projection instanceof ContentGraphProjectionInterface) {
+                    continue;
+                }
+                $projectionState = $projection->getState();
+                $this->projectionStateCache[$projectionState::class] = $projectionState;
+            }
+        }
```

Comment:
I don't completely get the implications of this, but why do we build up the runtime cache for all projections once any state is retrieved?

Reply:
because we did so previously almost as well just that we break on the first match, by building it once its more easy to grasp